src/wrappers/gcc_wrapper.cpp

```diff
@@ -86,6 +86,10 @@ string_list_t make_preprocessor_cmd(const string_list_t& args,
     drop_next_arg = false;
     if (arg == "-c") {
       drop_this_arg = true;
+    } else if (arg == "-MD") {
+      drop_this_arg = true;
+    } else if (arg == "-MMD") {
+      drop_this_arg = true;
```

Comment:
I believe that this change (while perfectly correct) will have a subtle side-effect: Old cache entries (cached before this change) will not contain .d files, so they rely on the .d-file being generated during the pre-processing step.

I think that the right solution is to tick the `HASH_VERSION` number so that there will be no cache hits for old cache entries with newer versions of BuildCache.

Reply:
Great catch.  Yes that is a good solution, will do.